eppo_client/client.py

```diff
@@ -36,143 +37,146 @@ def __init__(
             callback=config_requestor.fetch_and_store_configurations,
         )
         self.__poller.start()
+        self.__evaluator = Evaluator(sharder=MD5Sharder())
 
     def get_string_assignment(
-        self, subject_key: str, flag_key: str, subject_attributes=dict()
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[SubjectAttributes] = None,
+        default=None,
     ) -> Optional[str]:
-        try:
-            assigned_variation = self.get_assignment_variation(
-                subject_key, flag_key, subject_attributes, VariationType.STRING
-            )
-            return (
-                assigned_variation.typed_value
-                if assigned_variation is not None
-                else assigned_variation
-            )
-        except ValueError as e:
-            # allow ValueError to bubble up as it is a validation error
-            raise e
-        except Exception as e:
-            if self.__is_graceful_mode:
-                logger.error("[Eppo SDK] Error getting assignment: " + str(e))
-                return None
-            raise e
+        return self.get_assignment_variation(
+            subject_key,
+            flag_key,
+            subject_attributes,
+            VariationType.STRING,
+            default=default,
+        )
+
+    def get_integer_assignment(
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[SubjectAttributes] = None,
+        default=None,
+    ) -> Optional[float]:
+        return self.get_assignment_variation(
+            subject_key,
+            flag_key,
+            subject_attributes,
+            VariationType.INTEGER,
+            default=default,
+        )
+
+    def get_float_assignment(
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[SubjectAttributes] = None,
+        default=None,
+    ) -> Optional[float]:
+        return self.get_assignment_variation(
+            subject_key,
+            flag_key,
+            subject_attributes,
+            VariationType.FLOAT,
+            default=default,
+        )
 
+    @deprecated("get_numeric_assignment is deprecated in favor of get_float_assignment")
     def get_numeric_assignment(
```

Comment:
Not all SDKs literally use `Float`, some are `Double` - the usage of the word `numeric` is meant to be general purpose for numbers that have decimals.

Reply:
We can use `numeric` and `integer` if you prefer -- I changed it to float to make it clear that it is not an `integer`, which also sounds `numeric` to me

Reply:
I'm now convinced we should go with `numeric` (as in \mathbb{R}), and have every language represent that as best appropriate. And the other (new) type is `integer`. This makes the migration straightforward: we haven't changed the meaning of the existing type and there is no need to deprecate it; instead, we've added a specialized type that can be used from now on.